File: server/router/message.go
package router

import (
	"github.com/gin-gonic/gin"
	"server/api"
)

func MessageRouter(router *gin.Engine) {
	messageApi := api.MessageApi{}

	message := router.Group("api/message")
	{
		message.POST("/add", messageApi.AddMessage)
		message.GET("/list", messageApi.GetAllMessage)
		message.DELETE("/delete", messageApi.DeleteMessage)
		message.GET("/get", messageApi.GetMessageByAuthor)
		message.GET("/count", messageApi.MessageCount)
	}

}

File: web/src/api/message.ts
import request from "../utils/request";

export const AddMessage = (content: string) => request({
  url: '/message/add',
  method: 'post',
  data: {
    content
  }
})

export const GetAllMessage = () => request({
  url: '/message/list',
  method: 'get',
})

export const DeleteMessage = (id: number) => request({
  url: '/message/delete',
  method: 'delete',
  params: {
    id
  }
})

export const GetMessageByAuthor = () => request({
  url: '/message/get',
  method: 'get',
})

export const MessageCount = () => request({
  url: '/message/count',
  method: 'get'
})